url97a7a8bbec_PPilger_OOP_2_tgz-pJ8-TestJ8/src/Programm.java

```diff
@@ -0,0 +1,97 @@
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Programm {
+	private File file;
+	private List<Band> bands;
+
+	/**
+	 * Erzeugt ein neues Programm und laedt den letzten Zustand aus der Sicherungsdatei.
+	 */
+	public Programm() {
+		this(true);
+	}
+
+	/**
+	 * Erzeugt ein neues Programm
+	 * @param load true, wenn der letzte Zustand aus der Sicherung geladen werden soll.
+	 */
+	public Programm(boolean load) {
+		String filename = "bands.dat";
+		file = new File(filename);
+		
+		if (load) {
+			if (file.exists()) {
+				read();
+			}
+
+			if (this.bands == null) {
+				this.bands = new ArrayList<Band>();
+			}
+		} else {
+			this.bands = new ArrayList<Band>();
+		}
+	}
+
+	public void addBand(Band band) {
+		bands.add(band);
+	}
+
+	public Band getBand(String name) {
+		for (Band band : bands) {
+			if (band.getName().equalsIgnoreCase(name)) {
+				return band;
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * Speichert den aktuellen Programmzustand.
+	 */
+	public void quit() {
+		ObjectOutputStream dos = null;
+		try {
+			dos = new ObjectOutputStream(new FileOutputStream(file));
+			dos.writeObject(bands);
+		} catch (IOException e) {
+			e.printStackTrace();
+		} finally {
+			if (dos != null) {
+				try {
+					dos.close();
+				} catch (IOException e) {
+					e.printStackTrace();
+				}
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 66:72, 88:94.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.